src/adapters/poi/poi_store.js

```diff
@@ -30,4 +31,16 @@ export default class PoiStore extends Poi {
       return Object.assign(new PoiStore(), poi);
     });
   }
+
+  static deserialize(raw) {
+    const {
+      id, name, alternativeName, type, latLon, className, subClassName, bbox, blocks,
+      topImageUrl, kind,
+    } = raw;
+    if (kind === 'idunn') {
+      return new IdunnPoi(raw);
+    }
+    return new Poi(id, name, alternativeName, type, latLon, className, subClassName, bbox, blocks,
+      topImageUrl, raw);
```

Comment:
If I understand correctly your changes in the Poi class constructor, all `raw` fields will overwrite stuff that is explicitly pass as arguments. Maybe what you need is to destructure and pass only other unnamed fields. Like:
```js
const {
  id,
  name,
  // other named fields
  ...extras,
} = raw;
// ...
return new Poi(id, name, /* named fields */, extras);
```